docassemble/AssemblyLine/data/questions/al_document.yml

```diff
@@ -11,13 +11,46 @@ features:
 generic object: ALDocumentBundle
 event: x.send_email_action_event
 code: |
-  email_success = x.send_email(to=action_argument('email'), editable=action_argument('editable'), template=action_argument('template'))
+  email_success = x.send_email(
+      to=action_argument('email'),
+      editable=action_argument('editable'),
+      template=(
+        value(action_argument('template_name'))
+        if action_argument('template_name')
+        else None
+      ),
+      key=action_argument('key')
+  )
   if email_success:
     # Exact phrase below is in Docassemble's words dictionary
     log(word('E-mail was sent to') + ' ' + action_argument('email') ,'success')
   else:
     # E-mail failed is not in the default Docassemble words dictionary
     log(word('E-mail failed'), 'error')
+---
+generic object: ALDocumentBundle
+event: x.send_email_to_action_event
+code: |
+  email_success = x.send_email(
```

Comment:
I don't mind the duplication too much, but is it needed? this is the same as the previous event, except that it sends `json_response`; can we add `json_response` to the first event?

Reply:
The first event logs on screen, the second doesn't

Reply:
They both have the below lines that log on screen though:

```
  if email_success:
    log(word('E-mail was sent to') + ' ' + action_argument('email') ,'success')
    ... # an internal log in x.send_email_to_action_event
  else:
    log(word('E-mail failed'), 'error')
```

Reply:
One includes the email, the other doesn't. Actually the `log(...,"success")` didn't seem to work in the second event block with `json_response()` which I planned to investigate, but perhaps I should just disable it entirely as it doesn't currently do anything.

Reply:
Ah, gotcha. It read like it was doing the same thing, so removing it makes it a lot clearer what's happening.